microcontroller/Core/Src/main.c

```diff
@@ -667,82 +731,82 @@ static void MX_GPIO_Init(void)
 }
 
 /* USER CODE BEGIN 4 */
+//Only have access to GPIO pins in main
+void csv_header(char * file_name, int file_length) {
+	char name [file_length];
+	strcpy(name, file_name);
+	if (HAL_GPIO_ReadPin(GPIOB, GPIO_PIN_6)) {
+		file_res = sd_mount("/");
+
+		//Write to CSV using the following format for each column:
+		//sprintf(buffer, "<Column Title>,");
+		//update_file(name, buffer);
+
+		file_res = file_update(name, buffer);
+		sd_unmount("/");
+
+	}
+}
+
+void csv_update(char * file_name, int file_length) {
+
+	char name [file_length];
+	strcpy(name, file_name);
+
+	if (HAL_GPIO_ReadPin(GPIOB, GPIO_PIN_6)) {
+		file_res = sd_mount("/");
+		//Update CSV using following format:
+		//sprintf(buffer, "<%unit>, measurement name"
+		//update_file(name, buffer);
+
+
+
+		sprintf(buffer, "\n\n");
+		file_res = file_update(name, buffer);
+		sd_unmount("/");
+	}
+
+}
+
 void HAL_TIM_IC_CaptureCallback(TIM_HandleTypeDef *htim)
```

Comment:
You made some changes to this function. Are you able to change it back to the original code?